pkg/autoscaler/multiscaler.go

```diff
@@ -300,6 +304,10 @@ func (m *MultiScaler) RecordStat(key string, stat Stat) {
 	if exists {
 		logger := m.logger.With(zap.String(logkey.Key, key))
 		ctx := logging.WithLogger(context.TODO(), logger)
+
 		scaler.scaler.Record(ctx, stat)
+		if scaler.metric.Status.DesiredScale == 0 && stat.PodName == ActivatorPodName {
```

Comment:
Since we're scaling the activator, this won't work anymore I believe. You presumably need to do the same check as the autoscaler does, which is: `if strings.HasPrefix(stat.PodName, ActivatorPodName) {`

Has this worked locally?

Can the first half of this be: `scaler.getLatestScale() == 0`?

Reply:
Good find - I removed the pod name check entirely since it really shouldnt matter where the stat came from (as you point out).